api/src/main/java/org/pmiops/workbench/api/UserMetricsController.java

```diff
@@ -41,12 +48,19 @@
       userRecentResource -> {
         RecentResource resource = new RecentResource();
         resource.setCohort(TO_CLIENT_COHORT.apply(userRecentResource.getCohort()));
-        if (userRecentResource.getNotebookName() != null) {
-          FileDetail fileDetail = new FileDetail();
-          String[] notebookDetails = userRecentResource.getNotebookName().split(NOTEBOOKS_WORKSPACE_DIRECTORY);
-          fileDetail.setPath(notebookDetails[0] + NOTEBOOKS_WORKSPACE_DIRECTORY);
-          fileDetail.setName(notebookDetails[1]);
-          resource.setNotebook(fileDetail);
+        String notebookName = userRecentResource.getNotebookName();
+        if (notebookName != null && !notebookName.isEmpty()) {
+          try {
+            URI notebookUri = new URI(notebookName);
+            Path path = Paths.get(notebookUri.getPath());
+            String fileName = path.getFileName().toString();
+            String filePath = notebookName.replace(fileName, "");
```

Comment:
nit: To future proof, I'd make sure we only swap the last instacne of fileName here. e.g. if we wind up supporting subdirectories for notebooks or filenames without ipynb, 

gs://bkt/notebooks/asdf/asdf
gs://bkt/notebooks/foo.ipynb/foo.ipynb

Reply:
Gotcha - good catch.

Reply:
Done